### PR TITLE
T-50 - Refactor T-10 (PR)

### PR DESCRIPTION
**Pull request by AntonBessarabov**
- [x] ExchangeRateFactory - why this class is a factory?
- [x] Specific DateTime (see UnixTimeConverter) value must be in constants
- [x] In BankInfoPersister currencies dict should be in Enum
- [x] Method like a GetAndStoreExchangeRates - bad practice
- [x] BgpbInfoService almost the same as BsbInfoService - in this case you need to create an abstract class

- [ ] HttpWebRequest - legacy, There is an improved way to perform HTTP requests using by HttpClient with the async / await.
- [ ] You could create RequestProvider (for example) or a factory that would generate an HttpClient and then you could use it.
- [ ] Specific exception handling

Unsolved remarks moved to other tasks

### CODE REVIEW

**Comment by AntonBessarabov** [comment]
Please remember to check BankInfoPersister.cs

**Comment by AntonBessarabov** [comment]
[T-50 Refactor T-10 (PR)](https://trello.com/c/9GgwDoqn/48-t-50-refactor-t-10-pr)

File: PortfolioKeeper/BusinessLogic/Helpers/UnixTimeConverter.cs [hunk 1]
```diff
@@ -4,28 +4,26 @@ namespace PortfolioKeeper.BusinessLogic.Helpers
...
         {
             if (milliseconds < 0)
             {
                 throw new ArgumentOutOfRangeException(nameof(milliseconds), "Milliseconds should be >= 0");
             }
 
-            var dateTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, dateTimeKind);
-
-            return dateTime.AddMilliseconds(milliseconds);
+            return UnixTimeStartDate.AddMilliseconds(milliseconds);
```
[the remark this answers is not in the document]
Comment (AntonBessarabov) [reply PortfolioKeeper/BusinessLogic/Helpers/UnixTimeConverter.cs]: It's good idea to store date in local time?

File: PortfolioKeeper/BusinessLogic/Services/BankInfoService.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            IBankInfoProvider provider,
+            IBankInfoParser parser,
+            IBankInfoPersister bankInfoPersister)
+        {
+            this.provider = provider ?? throw new ArgumentNullException(nameof(provider));
+            this.parser = parser ?? throw new ArgumentNullException(nameof(parser));
+            this.bankInfoPersister = bankInfoPersister ?? throw new ArgumentNullException(nameof(bankInfoPersister));
+        }
+
+        public void GetExchangeRates()
```
[the remark this answers is not in the document]
Comment (AntonBessarabov) [reply PortfolioKeeper/BusinessLogic/Services/BankInfoService.cs]: Rename to PersistCurrentExchangeRates? Or ...?

File: PortfolioKeeper.Tests/BusinessLogic/Parsers/BgpbInfoParserTests.cs [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
+using System;
+using System.Collections.Generic;
 
 namespace PortfolioKeeper.Tests.BusinessLogic.Parsers
 {
     [TestFixture]
     public class BgpbInfoParserTests
     {
-        private Mock<IExchangeRateFactory> factory;
+        private Mock<IExchangeRateMapper> factory;
```
Comment (stas-litvinenka) [inline PortfolioKeeper.Tests/BusinessLogic/Parsers/BgpbInfoParserTests.cs]: factory->mapper

File: PortfolioKeeper.Tests/BusinessLogic/Factories/ExchangeRateFactoryTests.cs [hunk 4]
```diff
@@ -1,25 +1,25 @@
-﻿using System;
```
Comment (stas-litvinenka) [inline PortfolioKeeper.Tests/BusinessLogic/Factories/ExchangeRateFactoryTests.cs]: ExchangeRateFactoryTests -> ExchangeRateMapperTests

File: PortfolioKeeper/BusinessLogic/Services/BankInfoService.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            IBankInfoProvider provider,
+            IBankInfoParser parser,
+            IBankInfoPersister bankInfoPersister)
+        {
+            this.provider = provider ?? throw new ArgumentNullException(nameof(provider));
+            this.parser = parser ?? throw new ArgumentNullException(nameof(parser));
+            this.bankInfoPersister = bankInfoPersister ?? throw new ArgumentNullException(nameof(bankInfoPersister));
+        }
+
+        public void PersistCurrentExchangeRates()
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Services/BankInfoService.cs]: ProcessCurrentExchangeRates

**Pull request merged**